src/ReactPlaceholder.js

```diff
@@ -40,14 +40,20 @@ export default class ReactPlaceholder extends React.Component {
       type, customPlaceholder, showLoadingAnimation, ...rest
     } = this.props;
 
-    if (customPlaceholder) {
-      return customPlaceholder;
-    }
-
     const classes = showLoadingAnimation ?
       ['show-loading-animation', className].filter(c => c).join(' ') :
       className;
 
+    if (customPlaceholder && React.isValidElement(customPlaceholder)) {
+      const mergedCustomClasses =
+        [customPlaceholder.props.className, classes].filter(c => c).join(' ');
+      return React.cloneElement(customPlaceholder,
```

Comment:
does this work also with stateless components?
in case it doesn't we could simply return the `customPlaceholder` wrapper in a`div` and add the `className` to that `div`